src/bin/cut.rs

```diff
@@ -0,0 +1,525 @@
+    fn new(values: I, selected: slice::Iter<'a, bool>, to_eol: bool) -> Self {
+        SelectFromIter {
+            values: values,
+            selected: selected,
+            to_eol: to_eol,
+            consumed: false,
+        }
+    }
+}
+
+impl<'a, I> Iterator for SelectFromIter<'a, I> where I: Iterator
+{
+    type Item = I::Item;
+
+    fn next(&mut self) -> Option<I::Item> {
+        if self.consumed {
+            // The selected iterator was consumed ...
+            // but we continue yielding values from the values iterator.
+            return self.values.next();
+        } else {
+            // The selected iterator is not yet consumed.
+            // Consume both iterators and ..
+            // - If the value iterator fininishes, finish also SelectFromIter (return None).
+            // - If the value is selected, return it.
+            // - If the selected iterator finishes:
+            //      finish also SelectFromIter if not to_eol;
+            //      return the value otherwise and move to the other branch,
+            //        yield the rest from value iterator.
+            // - If the value is not selected (_), continue with the loop.
+            loop {
+                let (v, s) = (self.values.next(), self.selected.next());
+                if v.is_none() {
+                    return None;
+                }
+                match s {
+                    Some(&true) => return v,
+                    None => {
+                        // The selected iterator finished,
+                        if self.to_eol {
+                            self.consumed = true;
+                            return v;
+                        } else {
+                            return None;
+                        }
+                    }
+                    _ => {}
+                }
+            }
+        }
+        return None;
+    }
+}
+
+/// Set a boolean array element to true, extending the array if necessary.
+fn toggle(v: &mut Vec<bool>, index: usize) {
+    if index + 1 > v.len() {
+        v.resize(index + 1, false);
+    }
+    v[index] = true;
+}
+
+/// Fill a range (including first and last element) in a boolean array to true,
+/// extending the array if necessary.
+fn fill(v: &mut Vec<bool>, begin: usize, end: usize) {
+    // This could be more eficient by breaking it up into
+    // differente cases and filling with true and false
+    // but I do not think is worth it.
+    if end > v.len() {
+        v.resize(end, false);
+    }
+    for index in begin..end {
+        v[index] = true;
+    }
+}
+
+/// Write selected bytes from a reader.
+fn cut_bytes<R: Read, W: Write>(input: R, output: W, selection: &Selection) {
+    let reader = io::BufReader::new(input);
+    let mut writer = io::BufWriter::new(output);
+
+    for line in reader.lines() {
+        let line = line.unwrap();
+        let bytes = line.bytes();
+        for el in selection.select_from(bytes) {
+            let _ = writer.write(&[el]);
```

Comment:
You shouldn't fail silently. Please grep for `let _` and replace as many as possible with `try` from libextra.